### PR TITLE
add a view for Haskell STG intermediary output #3571

### PR DESCRIPTION
**Pull request by adamse**
image

### CODE REVIEW

**Comment by adamse** [comment]
A lot of unintended changes snuck in, i guess there is some formatting git-hook?

**Comment by mattgodbolt** [comment]
Hi @adamse - yes, there's a git hook which apparently still has some kinks in it. Please don't fret for now, we're trying to work out what the best course of action is. We may end up reformatting all the code, and then that will cause a bit of a problem for you. I'll be on hand to help if that is the case.

File: static/panes/compiler.js [hunk 1]
```diff
@@ -132,13 +132,19 @@ function Compiler(hub, container, state) {
         monacoDisassembly = languages[this.currentLangId].monacoDisassembly;
     }
 
-    this.outputEditor = monaco.editor.create(this.monacoPlaceholder[0], monacoConfig.extendConfig({
```
Comment (mattgodbolt) [inline static/panes/compiler.js:135]: Unfortunately like you said the auto format here has obscured the real change. Hoping if we reformat compiler.js in main and then you rebase, it'll revert to just being the diffs we'd expect.

**Review by mattgodbolt — COMMENTED** [review]
Looking good to me. Will see about reformatting the compiler.js so we can see the actual diff. Sorry for the delay on this, we'll get back to you soon.

**Comment by mattgodbolt** [comment]
Adam; can you try merging `main` and see if that goes well. If you have any issues I can give it a go.

**Comment by adamse** [comment]
Rebased! (and it still works when testing locally.)

**Comment by adamse** [comment]
I want to add 2 more views like this one: GHC Core and Cmm. It feels like there is opportunity to common up these extra output views, but I don't know what it would look like or if it is worth doing. Should I make an attempt or are we happy keeping the duplication?

**Comment by RubenRBS** [comment]
Historically we've been ok with duplication for the views, as there was no good way to properly share code that way. If by extending from a base class you can reduce some of this duplication and it's not _too_ hard to implement for you, go for it.

But! don't worry if you don't end up doing it that way :)

**Review by mattgodbolt — APPROVED** [review]
LGTM! Thanks so much! And +1 to @RubenRBS's response: we're still finding our feet with Typescript and what it can bring in terms of proper class hierarchies and code reuse, so anything's fine.

**Comment by RubenRBS** [comment]
This is now live

**Pull request merged**